components/canvas/webgl_mode/inprocess.rs

```diff
@@ -27,8 +28,10 @@ use webxr_api::SwapChainId as WebXRSwapChainId;
 pub struct WebGLComm {
     pub webgl_threads: WebGLThreads,
     pub webxr_swap_chains: SwapChains<WebXRSwapChainId>,
+    pub webxr_surface_providers: SurfaceProviders,
     pub image_handler: Box<dyn WebrenderExternalImageApi>,
     pub output_handler: Option<Box<dyn webrender_api::OutputImageHandler>>,
+    pub webgl_executor: WebGlExecutor,
```

Comment:
Scary API for running unknown code! I'm confused where this is used.

Reply:
This facilitates [this code](https://github.com/servo/webxr/pull/133/files#diff-f3cfd7c22eb8b9fb4aeefdabdb68f7e4R274-R288) which extracts the d3d11 device pointer from the surfman device on the webgl thread.

Reply:
OK, so as commented there if we could do this the other way round, and get the WebGL thread to use the OpenXR device, we could avoid this?

Reply:
The webgl thread is created at servo startup. The openxr device is created when a page requests an XR session.

Reply:
And I don't think we can't switch to using a different device later, because the page will have already created textures/framebuffers/etc. that won't be valid on a different device.

Reply:
Oh that's annoying! IIRC the spec does allow the GL context to be invalidated when the XR session starts, presumably for exactly this reason.